configuration/components/loki.libsonnet

```diff
@@ -652,6 +662,29 @@ function(params) {
       if std.length(config.resources) > 0
     },
 
+  local affinity(component, namespace) = {
+    podAntiAffinity: {
+      preferredDuringSchedulingIgnoredDuringExecution: [
+        {
+          weight: 100,
+          podAffinityTerm: {
+            topologyKey: 'kubernetes.io/hostname',
+            namespaces: [namespace],
```

Comment:
Can you elaborate a bit in the description on what happens if the target cluster will not have enough distinct nodes, e.g. for 3 ingesters schedule each on a different node?

Reply:
Here we use `preferredDuringSchedulingIgnoredDuringExecution`, which is a soft matcher and is ignored when unable to be fulfilled.